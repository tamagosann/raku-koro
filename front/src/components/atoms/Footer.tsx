import React, { FC } from 'react';
import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';
import MediaQuery from 'react-responsive';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    footer: {
      backgroundColor: '#fd7e00',
      color: '#fff',
      position: 'absolute',
      bottom: '0',
      width: '100%',
    },
    displayDevelopers: {
      textAlign: 'center',
    },
    flexContainer: {
      display: 'flex',
      justifyContent: 'space-between',
      alignItems: 'center',
      padding: '0 80px',
      maxWidth: '1200px',
      margin: '0 auto',
    },
    flexContainerColum: {
      display: 'flex',
      flexDirection: 'column-reverse',
      justifyContent: 'space-between',
      alignItems: 'center',
      padding: '0 80px',
    },
    flexItem: {
      display: 'flex',
      alignItems: 'center',
    },
    nameMargin: {
      marginRight: '16px',
    },
  })
);

export const Footer: FC = () => {
  const classes = useStyles();
  return (
    <footer id="footer" className={classes.footer} data-testid="footer-tag">
      <MediaQuery query="(min-width: 600px)">
        <div className={classes.flexContainer}>
          <p>&copy;{new Date().getFullYear()} Rakuraku Corona</p>
          <div>
            <h3 className={classes.displayDevelopers}>共同開発者</h3>
            <div>
              <div className={classes.flexItem}>
                <p className={classes.nameMargin}>M.T</p>
                <p>T.A</p>
              </div>
              <div className={classes.flexItem}>
                <p className={classes.nameMargin}>D.K</p>
                <p>A.S</p>
              </div>
              <div className={classes.flexItem}>
                <p className={classes.nameMargin}>T.K</p>
                <p>K.O</p>
              </div>
            </div>
          </div>
        </div>
      </MediaQuery>
      <MediaQuery query="(max-width: 599px)">
        <div className={classes.flexContainerColum}>
          <p>&copy;{new Date().getFullYear()} Rakuraku Corona</p>
          <div>
            <h3 className={classes.displayDevelopers}>共同開発者</h3>
            <div>
              <div className={classes.flexItem}>
                <p className={classes.nameMargin}>M.T</p>
                <p>T.A</p>
              </div>
              <div className={classes.flexItem}>
                <p className={classes.nameMargin}>D.K</p>
                <p>A.S</p>
              </div>
              <div className={classes.flexItem}>
                <p className={classes.nameMargin}>T.K</p>
                <p>K.O</p>
              </div>
            </div>
          </div>
        </div>
      </MediaQuery>
    </footer>
  );
};
