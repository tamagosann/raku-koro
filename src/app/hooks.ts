import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
import type { RootState, AppDispatch } from './store';

// useDispatchとかuseSelectorの代わりにこっちを使う
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
