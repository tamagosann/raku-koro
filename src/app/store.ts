import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import dailyInfectionRuducer from "../features/graphs/dailyInfectionSlice";
import prefectureRuducer from "../features/common/prefectureSlice";
import dailyDeadRuducer from "../features/graphs/dailyDeadSlice";
import totalCoronaRuducer from "../features/graphs/totalCoronaSlice";
import totalDethRuducer from "../features/graphs/totalDethSlice";
import vaccinationRuducer from "../features/graphs/vaccinationSlice";
import bedOccupancyRateRuducer from "../features/graphs/bedOccupancyRateSlice";
import dailyPositiveRuducer from "../features/graphs/dailyPositiveSlice";

export const store = configureStore({
  reducer: {
    dailyInfection: dailyInfectionRuducer,
    prefecture: prefectureRuducer,
    dailyDead: dailyDeadRuducer,
    totalCorona: totalCoronaRuducer,
    totalDeth: totalDethRuducer,
    vaccination: vaccinationRuducer,
    bedOccupancyRate: bedOccupancyRateRuducer,
    dailyPositive: dailyPositiveRuducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
