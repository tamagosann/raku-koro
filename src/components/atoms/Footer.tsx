import React, { FC } from 'react';
import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';
import MediaQuery from 'react-responsive';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    footer: {
      backgroundColor: '#fd7e00',
      color: '#fff',
      position: 'absolute',
      bottom: '0',
      width: '100%',
    },
    displayDevelopers: {
      textAlign: 'center',
    },
    flexContainer: {
      display: 'flex',
      justifyContent: 'space-between',
      alignItems: 'center',
      padding: '0 80px',
      maxWidth: '1200px',
      margin: '0 auto',
    },
    flexContainerColum: {
      display: 'flex',
      flexDirection: 'column-reverse',
      justifyContent: 'space-between',
      alignItems: 'center',
      padding: '0 80px',
    },
    flexItem: {
      display: 'flex',
      alignItems: 'center',
    },
    nameMargin: {
      marginRight: '16px',
    },
  })
);

export const Footer: FC = () => {
  const classes = useStyles();
  return (
    <footer id="footer" className={classes.footer} data-testid="footer-tag">
      <MediaQuery query="(min-width: 600px)">
        <div className={classes.flexContainer}>
          <p>&copy;{new Date().getFullYear()} Rakuraku Corona</p>
          <div>
            <h3 className={classes.displayDevelopers}>共同開発者</h3>
            <div>
              <div className={classes.flexItem}>
                <p className={classes.nameMargin}>豊山政一</p>
                <p>秋山太郎</p>
              </div>
              <div className={classes.flexItem}>
                <p className={classes.nameMargin}>工藤大輝</p>
                <p>坂本篤弥</p>
              </div>
              <div className={classes.flexItem}>
                <p className={classes.nameMargin}>春日崇希</p>
                <p>大谷走</p>
              </div>
            </div>
          </div>
        </div>
      </MediaQuery>
      <MediaQuery query="(max-width: 599px)">
        <div className={classes.flexContainerColum}>
          <p>&copy;{new Date().getFullYear()} Rakuraku Corona</p>
          <div>
            <h3 className={classes.displayDevelopers}>共同開発者</h3>
            <div>
              <div className={classes.flexItem}>
                <p className={classes.nameMargin}>豊山政一</p>
                <p>秋山太郎</p>
              </div>
              <div className={classes.flexItem}>
                <p className={classes.nameMargin}>工藤大輝</p>
                <p>坂本篤弥</p>
              </div>
              <div className={classes.flexItem}>
                <p className={classes.nameMargin}>春日崇希</p>
                <p>大谷走</p>
              </div>
            </div>
          </div>
        </div>
      </MediaQuery>
    </footer>
  );
};
