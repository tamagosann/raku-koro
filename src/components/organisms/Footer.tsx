import React, { FC } from "react";
import { makeStyles, createStyles, Theme } from "@material-ui/core/styles";
import MediaQuery from "react-responsive";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    footer: {
      backgroundColor: "#fd7e00",
      color: "#fff",
      position: "absolute",
      bottom: "0",
      width: "100%",
    },
    displayDevelopers: {
      textAlign: "center",
    },
    flexContainer: {
      display: "flex",
      justifyContent: "space-between",
      alignItems: "center",
      padding: "0 80px",
      maxWidth: "1200px",
      margin: "0 auto",
    },
    flexContainerColum: {
      display: "flex",
      flexDirection: "column-reverse",
      justifyContent: "space-between",
      alignItems: "center",
      padding: "0 80px",
    },
    flexItem: {
      display: "flex",
      alignItems: "center",
    },
    name: {
      margin: "10px",
    },
  })
);

const Footer: FC = () => {
  const classes = useStyles();
  return (
    <footer id="footer" className={classes.footer} data-testid="footer-tag">
      <MediaQuery query="(min-width: 600px)">
        <div className={classes.flexContainer}>
          <p>&copy;{new Date().getFullYear()} Rakuraku Corona</p>
          <div>
            <h3 className={classes.displayDevelopers}>共同開発者</h3>
            <div>
              <div className={classes.flexItem}>
                <p className={classes.name}>M.T</p>
                <p className={classes.name}>T.A</p>
              </div>
              <div className={classes.flexItem}>
                <p className={classes.name}>D.K</p>
                <p className={classes.name}>A.S</p>
              </div>
              <div className={classes.flexItem}>
                <p className={classes.name}>T.K</p>
                <p className={classes.name}>W.O</p>
              </div>
            </div>
          </div>
        </div>
      </MediaQuery>
      <MediaQuery query="(max-width: 599px)">
        <div className={classes.flexContainerColum}>
          <p>&copy;{new Date().getFullYear()} Rakuraku Corona</p>
          <div>
            <h3 className={classes.displayDevelopers}>共同開発者</h3>
            <div>
              <div className={classes.flexItem}>
                <p className={classes.name}>M.T</p>
                <p>T.A</p>
              </div>
              <div className={classes.flexItem}>
                <p className={classes.name}>D.K</p>
                <p>A.S</p>
              </div>
              <div className={classes.flexItem}>
                <p className={classes.name}>T.K</p>
                <p>W.O</p>
              </div>
            </div>
          </div>
        </div>
      </MediaQuery>
    </footer>
  );
};

export default Footer;
