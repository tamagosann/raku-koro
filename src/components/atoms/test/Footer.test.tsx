import React from 'react';
import {
  render,
  cleanup,
  screen,
  fireEvent,
  waitFor,
} from '@testing-library/react';
import { afterEach, expect } from '@jest/globals';
import { store } from '../../../app/store';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { Footer } from '../Footer';

describe('Footerコンポーネント', () => {
  // 各テストの直前に実行される（コンポーネントの取得）
  beforeEach(() => {
    render(
      <Provider store={store}>
        <Footer />
      </Provider>
    );
  });

  // 各処理が通った後にアンマウントさせる
  afterEach(() => cleanup());

  describe('jest renderer test', () => {
    test('render correctly', () => {
      const tree = renderer
        .create(
          <Provider store={store}>
            <Footer />
          </Provider>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('footerタグテスト', () => {
    test('fotterタグが存在するか否か', () => {
      const footerTag = screen.getByTestId('footer-tag');
      expect(footerTag).toBeInTheDocument();
    });
  });
});
