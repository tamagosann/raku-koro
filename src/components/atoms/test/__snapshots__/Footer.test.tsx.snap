// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footerコンポーネント jest renderer test render correctly 1`] = `
<footer
  className="makeStyles-footer-1"
  data-testid="footer-tag"
  id="footer"
/>
`;
